common/src/runners/mod.rs

```diff
@@ -188,11 +207,127 @@ impl BaseRunner {
         let tree_roots = hash_two::<Keccak256>(lt_tree_root, st_tree_root);
         Ok(tree_roots)
     }
+
+    pub fn get_lt_state(
+        &self, domain: &Domain, page_size: Option<usize>, next_token: Option<String>,
+    ) -> Result<LocalTrustStateResponse, Error> {
+        let rev_domain_indices = self
+            .rev_indices
+            .get(&domain.to_hash())
+            .ok_or::<Error>(Error::ReverseIndicesNotFound(domain.to_hash()))?;
+        let lt = self
+            .local_trust
+            .get(&domain.trust_namespace())
+            .ok_or(Error::LocalTrustNotFound(domain.trust_namespace()))?;
+
+        let lt_peers_cnt = lt.len() as u64;
+        let (from_peer_start, from_peer_end, to_peer_start, to_peer_end) =
+            compute_localtrust_peer_range(lt_peers_cnt, page_size, next_token)?;
```

Comment:
This logic assumes that we have a dense matrix where num_columns == num_rows.
We should assume that the matrix is sparse, i.e. the following shape:
```
0: [x1, x2, x3]
1: [y1, y2, y3, y4, y5, y6]
2: [z1, z2, z3, z4]
```

If we have a page size 6, we should return a vector:
```
[x1, x2, x3, y1, y2, y3]
```

Reply:
@lazovicff 
Here is update: https://github.com/openrankprotocol/openrank/pull/220/commits/5896a92dc886ccdd523ef6f773e9f969deac4f35.

I think it is difficult to track the correct `to_peer_index` in the `OutboundLocalTrust`, given `page_size` option.
The reason is that localtrust entry values are added or removed, and the `to_peer_index` is randomly inserted to underlying hashmap. (I mean Hashmap does not care the order of keys)
Hence, I think it is better to follow the seed trust fetch case.
I implement the logic of computing only `from_peer_index` from `next_token`.
Then, the querier gets the local trust entry data of certain peers(from_peer_start..from_peer_end).